File: src/com/timepath/hl2/io/bsp/lump/SurfaceEdgeHandler.java
package com.timepath.hl2.io.bsp.lump;

import com.timepath.hl2.io.bsp.Lump;
import com.timepath.hl2.io.bsp.LumpHandler;
import com.timepath.io.OrderedInputStream;
import java.io.IOException;
import java.util.logging.Logger;

class SurfaceEdgeHandler implements LumpHandler<int[]> {

    private static final Logger LOG = Logger.getLogger(SurfaceEdgeHandler.class.getName());

    public int[] handle(Lump l, OrderedInputStream in) throws IOException {
        int[] e = new int[l.length / 4];
        for(int i = 0; i < e.length; i++) {
            e[i] = in.readInt();
        }
        return e;
    }

}

File: src/main/kotlin/com/timepath/hl2/io/bsp/lump/SurfaceEdgeHandler.kt
package com.timepath.hl2.io.bsp.lump

import com.timepath.hl2.io.bsp.Lump
import com.timepath.hl2.io.bsp.LumpHandler
import com.timepath.io.OrderedInputStream
import java.io.IOException

class SurfaceEdgeHandler : LumpHandler<IntArray> {

    throws(IOException::class)
    override fun handle(l: Lump, `in`: OrderedInputStream): IntArray {
        val e = IntArray(l.length / 4)
        for (i in e.indices) {
            e[i] = `in`.readInt()
        }
        return e
    }

    companion object {

        private val MAX_MAP_SURFEDGES = 512000
    }
}
